### PR TITLE
PTP-1352 ERC1155 SDK Implementation

### PR DESCRIPTION
**Pull request by efecarranza**
# Changelog

Add ERC1155Mintable artifact.
Add ERC1155Mintable contract implementation.
Add error messages.
Add unit tests.

### CODE REVIEW

**Review by danielpavelicconsensys — APPROVED** [review]

File: integration-test/erc1155mintable.contract.test-integration.js [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+    const { addresses: addr, private_keys: pk } = require('./keys.json');
+    [owner, publicAddress, thirdUser] = Object.keys(addr);
+    const privateKey = pk[owner];
+    privateKeyPublicAddress = pk[publicAddress];
+
+    const rpcUrl = 'http://0.0.0.0:8545';
+    const chainId = 5;
+    const projectId = process.env.INFURA_PROJECT_ID;
+    const secretId = process.env.INFURA_PROJECT_SECRET;
+    const IPFS = { IPFSProjectID: '', IPFSProjectSecret: '' };
```
Comment (MqllR) [inline integration-test/erc1155mintable.contract.test-integration.js:33]: Not used

File: integration-test/erc1155mintable.contract.test-integration.js [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+    expect(infos).toStrictEqual([utils.getAddress(publicAddress), BigNumber.from('1')]);
+  });
+
+  it('should renounce contract ownership', async () => {
+    const result = await contractObject.accessControl.renounceOwnership();
+    const receipt = await result.wait();
+
+    expect(receipt.status).toBe(1);
+  });
+});
```
Comment (salimtb) [inline integration-test/erc1155mintable.contract.test-integration.js:175]: i think its better to follow the same logic and split the integration tests , we can create folders for each template

File: integration-test/erc1155mintable.contract.test-integration.js [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+
+    await txApprove.wait();
+
+    const accountPublic = new Auth({
+      privateKey: privateKeyPublicAddress,
+      projectId: process.env.INFURA_PROJECT_ID,
+      secretId: process.env.INFURA_PROJECT_SECRET,
+      rpcUrl: 'http://0.0.0.0:8545',
+      chainId: 5,
+    });
```
Comment (MqllR) [inline integration-test/erc1155mintable.contract.test-integration.js:133]: Exactly the same than `account` no ?

File: src/lib/ContractTemplates/ERC1155Mintable.js [hunk 4]
```diff
@@ -0,0 +1,603 @@
...
+        errorLogger({
+          location: ERROR_LOG.location.ERC1155Mintable_mintBatch,
+          message: ERROR_LOG.message.an_error_occured,
+          options: `${type} ${message}`,
+        }),
+      );
+    }
+  }
+
+  async addIds({ ids = [], gas = null }) {
```
Comment (MqllR) [inline src/lib/ContractTemplates/ERC1155Mintable.js:259]: Missing method documentation

File: src/lib/ContractTemplates/ERC1155Mintable.js [hunk 5]
```diff
@@ -0,0 +1,603 @@
...
+  }
+
+  /**
+   * setContractURI function: Set the "contractURI" metadata for the specified contract
+   * @param {string} contractURI ContractURI for the contract
+   * (URI to a JSON file describing the contract's metadata)
+   * @notice Warning: This method will consume gas (35000 gas estimated)
+   * @returns {Promise<ethers.providers.TransactionResponse>} Transaction
+   */
+  async setContractURI({ contractURI, gas = null }) {
```
Comment (MqllR) [inline src/lib/ContractTemplates/ERC1155Mintable.js:346]: Missing gas attribute in the documentation

File: src/lib/ContractTemplates/ERC1155Mintable.js [hunk 6]
```diff
@@ -0,0 +1,603 @@
...
+  /**
+   * SafeTransfer function: safeTransfer the token 'tokenId' between 'from' and 'to addresses.
+   * @param {string} from Address who will transfer the token
+   * @param {string} to Address that will receive the token
+   * @param {number} tokenId ID of the token that will be transfered
+   * @param {number} quantity quantity of the given tokenId to be transferred
+   * @notice Warning: This method will consume gas (xxx gas estimated)
+   * @returns {Promise<ethers.providers.TransactionResponse>} Transaction
+   */
+  async transfer({ from, to, tokenId, quantity, gas = null }) {
```
Comment (MqllR) [inline src/lib/ContractTemplates/ERC1155Mintable.js:395]: Same, `gas` is not documented

File: src/lib/ContractTemplates/ERC1155Mintable.js [hunk 7]
```diff
@@ -0,0 +1,603 @@
...
+  /**
+   * TransferBatch function: safeBatchTransfer the 'tokenIds' between 'from' and 'to addresses.
+   * @param {string} from Address who will transfer the token
+   * @param {string} to Address that will receive the token
+   * @param {number} tokenIds IDs of the tokens that will be transferred
+   * @param {number} quantities quantities of the given tokenId to be transferred
+   * @notice Warning: This method will consume gas (xxx gas estimated)
+   * @returns {Promise<ethers.providers.TransactionResponse>} Transaction
+   */
+  async transferBatch({ from, to, tokenIds, quantities, gas = null }) {
```
Comment (MqllR) [inline src/lib/ContractTemplates/ERC1155Mintable.js:466]: same

File: src/lib/ContractTemplates/ERC1155Mintable.js [hunk 8]
```diff
@@ -0,0 +1,603 @@
...
+      );
+    }
+  }
+
+  /**
+   * Load an ERC1155Mintable contract from an existing contract address. Used by the SDK class
+   * @param {string} contractAddress Address of the ERC1155Mintable contract to load
+   * @returns void
+   */
+  async loadContract({ contractAddress }) {
```
Comment (MqllR) [inline src/lib/ContractTemplates/ERC1155Mintable.js:110]: I'd love to get the returned object of `ethers.Contract` for this method =)

**Review by MqllR — APPROVED** [review]

**Pull request merged**